src/sentry/api/serializers/models/sentry_app.py

```diff
@@ -31,10 +32,13 @@ def serialize(self, obj, attrs, user):
         if is_active_superuser(env.request) or (
             hasattr(user, "get_orgs") and obj.owner in user.get_orgs()
         ):
+            client_secret = (
+                obj.application.client_secret if obj.show_auth_info(access) else MASKED_VALUE
```

Comment:
@MeredithAnya It might be nice to push this into a `get_visible_client_secret` method on the sentry app so you don't have to import `MASKED_VALUE`.